game-core/src/main/java/games/strategy/triplea/ui/UnitChooser.java

```diff
@@ -418,7 +418,9 @@ void disableMax() {
       defaultHits = new ArrayList<>(Math.max(1, category.getHitPoints() - category.getDamaged()));
       final int numUnits = category.getUnits().size();
       int hitsUsedSoFar = 0;
-      for (int i = 0; i < Math.max(1, category.getHitPoints() - category.getDamaged()); i++) {
+      for (int i = 0, m = Math.max(1, category.getHitPoints() - category.getDamaged());
+          i < m;
+          i++) {
```

Comment:
Actually I think it would be better to just move initialisation and deklaration out of the loop, up one line

Reply:
IE, convert the for loop to a while loop?

Reply:
Not really, just the constant variable 'm'

Reply:
That has a problem where the `m` gets increased scoping compared to what's needed.

FWIW, computing loop termination conditions like this used to be quite idiomatic. PMD used to recommend it. This performance improvement I think just became less common with Java 1.5 and for-each loops as there are usually better ways to iterate and iterating with a non-constant loop condition is less common.